core/src/mender-deployment-data.c

```diff
@@ -108,7 +108,7 @@ mender_set_deployment_data(mender_deployment_data_t *deployment_data) {
     assert(NULL != item); /* Validation above should have catched this already */
     if (MAX_STATE_DATA_STORE_COUNT <= cJSON_GetNumberValue(item)) {
```

Comment:
In this task, we also need to get a sensible value for `MAX_STATE_DATA_STORE_COUNT`. We should document (in the code) why we chose such a number.

In the C++ project we have this:
```
// The maximum times we are allowed to move through update states. If this is exceeded then the
// update will be forcefully aborted. This can happen if we are in a reboot loop, for example.
const int kMaxStateDataStoreCount = 28;
```

I'll ask Kristian about the guideline for this number.

Reply:
From the [commit](https://github.com/mendersoftware/mender/commit/12e6c62302e6dcca51d3efb382e276515552b718) that added it: 
`There is nothing accurate about this number. It's just a number which
won't loop too many times, but still gives some wiggle room to add a
few more states without exceeding the number. It was determined simply
by matching the number of loops in one of the tests which tries to
exceed it.`